tony-history-server/app/hadoop/Requirements.java

```diff
@@ -41,25 +50,57 @@ private void createDirIfNotExists(FileSystem myFs, Path dir, FsPermission perm)
     }
   }
 
-  @Inject
-  public Requirements(Config appConf) {
-    HdfsConfiguration conf = Configuration.getHdfsConf();
-    FileSystem myFs = HdfsUtils.getFileSystem(conf);
-    if (myFs == null) {
-      LOG.error("Failed to initialize file system");
-      return;
+  private void initFs(HdfsConfiguration conf) {
+    try {
+      histFs = histFolder.getFileSystem(conf);
+    } catch (IOException e) {
+      LOG.error("Encountered exception while getting filesystem for " + histFolder, e);
     }
+  }
+
+  private void setUpKeytab(HdfsConfiguration hdfsConf) {
```

Comment:
minor: setup, to be consistent with below

Reply:
Fixed